File: app/src/main/java/com/tomaschlapek/tcbasearchitecture/widget/SquareImageView.java
package com.tomaschlapek.tcbasearchitecture.widget;

import android.content.Context;
import android.util.AttributeSet;

/**
 * Created by tomaschlapek on 28/7/17.
 */

public class SquareImageView extends android.support.v7.widget.AppCompatImageView {

  public SquareImageView(Context context) {
    super(context);
  }

  public SquareImageView(Context context, AttributeSet attrs) {
    super(context, attrs);
  }

  public SquareImageView(Context context, AttributeSet attrs, int defStyle) {
    super(context, attrs, defStyle);
  }

  @Override
  protected void onMeasure(int widthMeasureSpec, int heightMeasureSpec) {
    super.onMeasure(widthMeasureSpec, heightMeasureSpec);
    setMeasuredDimension(getMeasuredWidth(), getMeasuredWidth()); //Snap to width
  }
}

File: app/src/main/kotlin/com/tomaschlapek/tcbasearchitecture/widget/SquareImageView.kt
package com.tomaschlapek.tcbasearchitecture.widget

import android.content.Context
import android.util.AttributeSet


class SquareImageView : android.support.v7.widget.AppCompatImageView {

  constructor(context: Context) : super(context) {}

  constructor(context: Context, attrs: AttributeSet) : super(context, attrs) {}

  constructor(context: Context, attrs: AttributeSet, defStyle: Int) : super(context, attrs, defStyle) {}

  override fun onMeasure(widthMeasureSpec: Int, heightMeasureSpec: Int) {
    super.onMeasure(widthMeasureSpec, heightMeasureSpec)
    setMeasuredDimension(measuredWidth, measuredWidth) //Snap to width
  }
}
